pkg/controller/appdefinition/acorn.go

```diff
@@ -77,19 +78,29 @@ func scopeLinks(app *v1.AppInstance, bindings v1.ServiceBindings) (result v1.Ser
 }
 
 func toAcorn(appInstance *v1.AppInstance, tag name.Reference, pullSecrets *PullSecrets, acornName string, acorn v1.Acorn) *v1.AppInstance {
-	image := images.ResolveTag(tag, acorn.Image)
-	if strings.HasPrefix(acorn.Image, "sha256:") {
-		image = strings.TrimPrefix(acorn.Image, "sha256:")
+	var image string
+	_, isPattern := autoupgrade.AutoUpgradePattern(acorn.Image)
+	isPattern = isPattern && acorn.AutoUpgrade != nil && *acorn.AutoUpgrade
```

Comment:
An auto-upgrade pattern is enough for the app to be considered an "auto upgraded app"

Reply:
A user could still theoretically create a nested Acorn with a pattern in the image but have forgotten to set the auto upgrade parameter though, so I figured it would be best to explicitly check for that as well.

Reply:
I am saying that a user doesn't need to explicitly set `AutoUpgrade` on a nested acorn. Using an auto-upgrade pattern is enough to say that auto-upgrade is on.

The logic, as it is now, requires both a pattern and explicitly setting `AutoUpgrade`.

Reply:
Oh I did not realize that we infer that auto-upgrade is on based on finding the pattern. Fixed.